python/tests/unit/arcticdb/version_store/test_parallel.py

```diff
@@ -240,3 +242,157 @@ def test_datetimes_to_nats(lmdb_version_store):
         result = result.astype(df.dtypes)
 
     assert_frame_equal(result, df)
+
+
+@pytest.mark.parametrize("append", (True, False))
+@pytest.mark.parametrize("arg", (True, False, None))
+@pytest.mark.parametrize("lib_config", (True, False))
+def test_compact_incomplete_prune_previous(lib_config, arg, append, version_store_factory):
+    lib = version_store_factory(prune_previous_version=lib_config)
+    lib.write("sym", pd.DataFrame({"col": [3]}, index=pd.DatetimeIndex([0])))
+    lib.append("sym", pd.DataFrame({"col": [4]}, index=pd.DatetimeIndex([1])), incomplete=True)
+
+    lib.compact_incomplete("sym", append, convert_int_to_float=False, prune_previous_version=arg)
+    assert lib.read_metadata("sym").version == 1
+
+    should_prune = lib_config if arg is None else arg
+    assert lib.has_symbol("sym", 0) != should_prune
+
+
+def test_compact_incomplete_sets_sortedness(lmdb_version_store):
+    lib = lmdb_version_store
+    sym = "test_compact_incomplete_sets_sortedness"
+    df_0 = pd.DataFrame({"col": [1, 2]}, index=[pd.Timestamp("2024-01-01"), pd.Timestamp("2024-01-02")])
+    df_1 = pd.DataFrame({"col": [3, 4]}, index=[pd.Timestamp("2024-01-03"), pd.Timestamp("2024-01-04")])
+    lib.write(sym, df_1, parallel=True)
+    lib.write(sym, df_0, parallel=True)
+    lib.compact_incomplete(sym, False, False)
+    assert lib.get_info(sym)["sorted"] == "ASCENDING"
+
+    df_2 = pd.DataFrame({"col": [5, 6]}, index=[pd.Timestamp("2024-01-05"), pd.Timestamp("2024-01-06")])
+    df_3 = pd.DataFrame({"col": [7, 8]}, index=[pd.Timestamp("2024-01-07"), pd.Timestamp("2024-01-08")])
+    lib.append(sym, df_3, incomplete=True)
+    lib.append(sym, df_2, incomplete=True)
+    lib.compact_incomplete(sym, True, False)
+    assert lib.get_info(sym)["sorted"] == "ASCENDING"
+
+
+@pytest.mark.parametrize("append", (True, False))
+def test_parallel_sortedness_checks(lmdb_version_store, append):
+    lib = lmdb_version_store
+    sym = "test_parallel_sortedness_checks"
+    if append:
```

Comment:
Lots of the tests from here down are substantially different depending on `append`. I think it would be clearer to explode out to separate tests rather than parameterizing.

Reply:
I disagree on this. The core features under test are identical for `write` and `append`. Having them parametrized makes it clear that they should behave the same way.